pkg/jobs/mem_jobs_test.go

```diff
@@ -184,6 +184,33 @@ func TestRetry(t *testing.T) {
 	w.Wait()
 }
 
+func TestPanic(t *testing.T) {
+	var w sync.WaitGroup
+
+	maxExecCount := 4
+
+	broker := NewMemBroker("panic", WorkersList{
+		"panic": {
+			Concurrency:  1,
+			MaxExecCount: uint(maxExecCount),
+			RetryDelay:   1 * time.Millisecond,
+			WorkerFunc: func(ctx context.Context, _ *Message) error {
+				w.Done()
+				panic("oops")
+			},
+		},
+	})
+
+	w.Add(maxExecCount)
+	_, _, err := broker.PushJob(&JobRequest{
+		WorkerType: "panic",
+		Message:    nil,
+	})
+
+	assert.NoError(t, err)
+	w.Wait()
```

Comment:
What I'd have done is a worker that panics when given an odd number and works fine when given an even number, then give it a job with an odd number, and then another job with an even number, and check that the even number has been correctly processed (ie the panic on the odd number has not interrupted the worker). This doesn't mean that your test is wrong, just that I don't understand it.